### ISSUE TITLE
🐛 Biome encountered an unexpected error

### ISSUE THREAD

**Issue by 0xtz** [biomejs/biome#11024]
### Environment information

<details>

```bash

RUST_BACKTRACE=1 biome format

```

</details>

### What happened?

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_module_graph/src/js_module_info/module_resolver.rs:500:33
Thread Name: biome::workspace_worker_11
Message: index out of bounds: the len is 107 but the index is 287
Stack Trace: Re-run with `RUST_BACKTRACE=1` to capture the stack trace

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_module_graph/src/js_module_info/module_resolver.rs:500:33
Thread Name: biome::workspace_worker_1
Message: index out of bounds: the len is 62 but the index is 287
Stack Trace: Re-run with `RUST_BACKTRACE=1` to capture the stack trace

src/components/admin/products/category-management-sheet.tsx internalError/panic  INTERNAL  ━━━━━━━━━━

  ✖ processing panicked: index out of bounds: the len is 107 but the index is 287

  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.

src/components/ui/search-input.tsx internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: index out of bounds: the len is 62 but the index is 287

  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
                                                                                         Checked 120 files in 390ms. No fixes applied.
[tz:~/innovstay/lathyou]$ RUST_BACKTRACE=1 bun run fix         (feat/admin-proxy-auth✱)
$ ultracite fix
Biome encountered an unexpected error
                                                                                         This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.
                                                                                         When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:
                                                                                         Source Location: crates/biome_module_graph/src/js_module_info/module_resolver.rs:500:33  Thread Name: biome::workspace_worker_13
Message: index out of bounds: the len is 107 but the index is 287
Stack Trace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>                                                                             7: <unknown>                                                                             8: <unknown>
   9: <unknown>
  10: <unknown>                                                                            11: <unknown>                                                                            12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>                                                                            20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>                                                                            25: start_thread
             at ./nptl/pthread_create.c:447:8
  26: clone3
             at ./misc/../sysdeps/unix/sysv/linux/x86_64/clone3.S:78:0

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_module_graph/src/js_module_info/module_resolver.rs:500:33
Thread Name: biome::workspace_worker_3
Message: index out of bounds: the len is 62 but the index is 287
Stack Trace:
   0: <unknown>                                                                             1: <unknown>                                                                             2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: start_thread
             at ./nptl/pthread_create.c:447:8
  26: clone3
             at ./misc/../sysdeps/unix/sysv/linux/x86_64/clone3.S:78:0

src/components/admin/products/category-management-sheet.tsx internalError/panic  INTERNAL  ━━━━━━━━━━

  ✖ processing panicked: index out of bounds: the len is 107 but the index is 287

  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.

src/components/ui/search-input.tsx internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: index out of bounds: the len is 62 but the index is 287

  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.

Checked 120 files in 398ms. No fixes applied.

### Expected result

-

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Update to the latest. We fixed this bug
